src/Polysemy/Final/Async.hs

```diff
@@ -18,21 +20,42 @@ import Polysemy.Final
 --
 -- /Beware/: Effects that aren't interpreted in terms of 'IO'
 -- will have local state semantics in regards to 'Async' effects
--- interpreted this way. See 'interpretFinal'.
+-- interpreted this way. See 'Final'.
 --
 -- Notably, unlike 'asyncToIO', this is not consistent with
--- 'Polysemy.State.State' unless 'Polysemy.State.runStateInIORef' is used.
+-- 'Polysemy.State.State' unless 'Polysemy.State.runStateIORef' is used.
 -- State that seems like it should be threaded globally throughout the `Async`
 -- /will not be./
 --
--- Prefer 'asyncToIO' unless its unsafe or inefficient in the context of your
--- application.
-runAsyncFinal :: Member (Final IO) r
-              => Sem (Async ': r) a
-              -> Sem r a
-runAsyncFinal = interpretFinal $ \case
+-- Prefer 'asyncToIO' or 'asyncToIOThreadless' unless these are unsafe or
```

Comment:
I know you have the comment above about `lowerAsync`, but I think it warrants suggesting to prefer this over `lowerAsync`; my understanding is we can deprecate that thing when this lands?

Reply:
The only thing I'm worried about is if `-Final` interpreters somehow perform worse. (I don't see how, but I haven't checked). Otherwise, yeah, `lowerAsync`, `lowerError`, `lowerResource`, and `runFixpointM` should all be safe to deprecate.

Reply:
They're questionable enough that we should just deprecate them and deal with perf if it comes to that.

Reply:
Ok, I'll update the comments to say "prefer `-Final` interpreters".

Reply:
Thanks!